app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,92 @@
+class TweetsController < ApplicationController
+
+#shows tweets index page, showing all tweets
+  get '/tweets' do
+    if logged_in? #if user is logged in, show all tweets in database via index view
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    else
+      redirect '/login' #if user is not logged in, show login page
+    end
+  end
+
+#show form to create new tweet
+  get '/tweets/new' do
+    if logged_in?  #if logged in, show the new tweet form via new view
+      erb :'tweets/new'
+    else
+      redirect '/login' #if not logged in, show login screen
+    end
+  end
+
+  #show a specific tweet given its ID
+    get '/tweets/:id' do
+      if logged_in? #if logged in...
+        @tweet = Tweet.find_by_id(params[:id]) #find specific tweet by ID and assign to @tweet
+        erb :'/tweets/show' #show content in @tweet via show view
+      else
+        redirect '/login'
+      end
+    end
+
+#process new tweet form submission based on params
+  post '/tweets' do
+    if logged_in? #if logged in...
+      if !params[:content].empty? #and if tweet_content isn't empty...
+        @tweet = current_user.tweets.build(params) #create new tweet instance via #build
+        @tweet.save #save new tweet instance to database
+          redirect "/tweets/#{@tweet.id}" #get taken to the tweet's page
+      else
+        redirect '/tweets/new' #if tweet content is empty, go back to new tweet form page
+      end
+    else
+      redirect '/login' #if not logged in, go to login page
+    end
+  end
+
+#show edit tweet page for a specific tweet
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user == current_user
```

Comment:
We could eliminate some lookup time here by only searching the current user's tweets, and then use just @tweet to control the following `if`:
```
@tweet = current_user.tweets.find_by(id: params[:id])
if @tweet
  # ...
```
This comment also applies to the patch and delete routes.